cli-plugins/manager/cobra.go

```diff
@@ -90,11 +90,12 @@ func AddPluginCommandStubs(dockerCli command.Cli, rootCmd *cobra.Command) (err e
 					cargs = append(cargs, args...)
 					cargs = append(cargs, toComplete)
 					os.Args = cargs
-					runCommand, runErr := PluginRunCommand(dockerCli, p.Name, cmd)
+					pluginRunCommand, runErr := PluginRunCommand(dockerCli, p.Name, cmd)
 					if runErr != nil {
 						return nil, cobra.ShellCompDirectiveError
 					}
-					runErr = runCommand.Run()
+					runCommand := dockerCli.(*command.DockerCli).InstrumentPluginCommand(pluginRunCommand)
```

Comment:
Haven't reviewed this properly yet, so sorry if I'm wrong, but couldn't the `InstrumentPluginCommand` be a part of the `TelemetryClient` interface, so we don't need a typecast?

Reply:
Yep you're right, good catch

Reply:
Ah, I gave it a little bit more attention and looks like it will also expose the `InstrumentPluginCommand` (which returns a private struct) to all CLI plugins. Which we probably don't want to, as no plugin should really call this method on its own.

Keeping it internal and having a typecast would probably be better indeed 🙈 I think we should get back to the previous version with a type cast, and just add a check if the cast succeeded so we don't panic in some weird edge case.

Sorry for the noise 🙈